src/ReaderScreen/reader.jsx

```diff
@@ -1,238 +1,47 @@
-import { View, StatusBar, Text, StyleSheet, ScrollView } from "react-native";
+import React, { useEffect, useState, useRef, useCallback } from "react";
+import { View, StatusBar, ScrollView, ActivityIndicator } from "react-native";
 import PropTypes from "prop-types";
-import { Icon } from "@rneui/themed";
 import { SafeAreaProvider, SafeAreaView } from "react-native-safe-area-context";
-import { getShabadFromID } from "../database/db";
 import colors from "../common/colors";
 import constant from "../common/constant";
-import { fontSizeForReader, fontColorForReader } from "./utils/util";
 import { setBookmarkPosition } from "../common/actions";
-
-// const { height, width } = Dimensions.get("window");
+import ShabadItem from "./components/shabadItem";
+import AutoScrollComponent from "./components/autoScrollComponent";
+import Header from "./components/header";
+import { useFetchShabad, usePagination } from "./utils/hooks";
 
 function Reader({ navigation, route }) {
-  const readerRef = useRef();
+  const readerRef = useRef(null);
 
-  const {
-    baniLength,
-    transliterationLanguage,
-    isNightMode,
-    fontFace,
-    fontSize,
-    larivaar,
-    isEnglishTranslation,
-    isPunjabiTranslation,
-    isSpanishTranslation,
-    isTransliteration,
-    bookmarkPosition,
-  } = useSelector((state) => state);
-  const [shabadID, setShabadID] = useState(Number(route.params.params.id));
+  const { isNightMode, bookmarkPosition, isAutoScroll } = useSelector((state) => state);
+  const [shabadID] = useState(Number(route.params.params.id));
   const [rowHeights, setRowHeights] = useState([]);
-  const [shabad, setShabad] = useState([]);
-  const [page, setPage] = useState([]);
-  const [endReached, setEndReached] = useState(false);
-  const [bookmarkIndex, setBookmarkIndex] = useState(0);
-  const [maxPageLength, setMaxPageLength] = useState(50);
+  const [itemsCount, setItemsCount] = useState(50);
   const dispatch = useDispatch();
+  const { shabad, isLoading } = useFetchShabad(shabadID);
+  const { currentPage, handleScroll } = usePagination(shabad, itemsCount);
 
-  const pagination = (data) => {
-    setPage(data.slice(0, maxPageLength));
-  };
-
-  useEffect(() => {
-    (async () => {
-      try {
-        setShabadID(Number(route.params.params.id));
-        const data = await getShabadFromID(shabadID, baniLength, transliterationLanguage);
-        setShabad(data);
-        pagination(data);
-      } catch (error) {
-        console.log("Error eh wala ", error);
-      }
-    })();
-  }, [shabadID, transliterationLanguage, bookmarkPosition]);
-  const headerLeft = () => (
-    <Icon
-      name="arrow-back"
-      size={30}
-      onPress={() => navigation.goBack()}
-      color={colors.WHITE_COLOR}
-    />
+  const handleBackPress = useCallback(() => navigation.goBack(), [navigation]);
+  const handleBookmarkPress = useCallback(() => {
+    navigation.navigate("Bookmarks", { id: shabadID });
```

Comment:
veerji, Please move all the routes to a constant routes js file.

Reply:
veer ji I tried separating but that's the best way I can get here in react-native.